File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            n = nums[i]
            if n in d:
                d[n].append(i)
            else:
                d[n] = [i]
        
        for k in d:
            complement = target - k
            if complement != k and complement in d:
                return [d[k][0], d[complement][0]]
            elif complement == k and len(d[k]) >= 2:
                return [d[k][0], d[k][1]]
        

            

File: 1-two-sum/two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> positions = new HashMap<>();
        for (int i = 0; i< nums.length; i++) {
            int n = nums[i];
            positions.put(n, i);
        }
        int[] res = new int[2];

        for (int i = 0; i< nums.length; i++) {
            int complement = target - nums[i];
            if (positions.containsKey(complement) && i != positions.get(complement)) {
                res[0] = i;
                res[1] = positions.get(complement);
            }
        }

        return res;
    }
    

}